### ISSUE TITLE
False positive for PLW2901 with annotated assignment without value

### ISSUE THREAD

**Issue by dhruvmanila** [charliermarsh/ruff#3888]
### Environment
* `ruff --version`: `0.0.261`

### Code (`t.py`):
```python
for i in []:
    i: int  # This is an annotated assignment without value
```

### Running `ruff`:
```console
$ ruff check --isolated --show-source --no-cache --select PLW2901 t.py 
t.py:2:5: PLW2901 `for` loop variable `i` overwritten by assignment target
  |
2 |     i: int  # no error
  |     ^ PLW2901
  |

Found 1 error.
```
